src/main/java/com/box/sdk/BoxGroupMembership.java

```diff
@@ -0,0 +1,235 @@
+
+    /**
+     * Constructs a BoxGroupMembership for a group membership with a given ID.
+     * @param  api the API connection to be used by the group membership.
+     * @param  id  the ID of the group membership.
+     */
+    public BoxGroupMembership(BoxAPIConnection api, String id) {
+        super(api, id);
+    }
+
+    /**
+     * Gets information about this group membership.
+     * @return info about this group membership.
+     */
+    public Info getInfo() {
+        BoxAPIConnection api = this.getAPI();
+        URL url = MEMBERSHIP_URL_TEMPLATE.build(api.getBaseURL(), this.getID());
+
+        BoxAPIRequest request = new BoxAPIRequest(api, url, "GET");
+        BoxJSONResponse response = (BoxJSONResponse) request.send();
+        JsonObject jsonObject = JsonObject.readFrom(response.getJSON());
+        return new Info(jsonObject);
+    }
+
+    /**
+     * Updates the information about this group membership with any info fields that have been modified locally.
+     * @param info the updated info.
+     */
+    public void updateInfo(Info info) {
+        BoxAPIConnection api = this.getAPI();
+        URL url = MEMBERSHIP_URL_TEMPLATE.build(api.getBaseURL(), this.getID());
+
+        BoxJSONRequest request = new BoxJSONRequest(api, url, "PUT");
+        request.setBody(info.getPendingChanges());
+        BoxJSONResponse response = (BoxJSONResponse) request.send();
+        JsonObject jsonObject = JsonObject.readFrom(response.getJSON());
+        info.update(jsonObject);
+    }
+
+    /**
+     * Deletes this group membership.
+     */
+    public void delete() {
+        BoxAPIConnection api = this.getAPI();
+        URL url = MEMBERSHIP_URL_TEMPLATE.build(api.getBaseURL(), this.getID());
+
+        BoxAPIRequest request = new BoxAPIRequest(api, url, "DELETE");
+        BoxAPIResponse response = request.send();
+        response.disconnect();
+    }
+
+    /**
+     * Contains information about a BoxGroupMembership.
+     */
+    public class Info extends BoxResource.Info {
+        private BoxUser.Info user;
+        private BoxGroup.Info group;
+        private Role role;
+        private Date createdAt;
+        private Date modifiedAt;
+
+        /**
+         * Constructs an empty Info object.
+         */
+        public Info() {
+            super();
+        }
+
+        /**
+         * Constructs an Info object by parsing information from a JSON string.
+         * @param  json the JSON string to parse.
+         */
+        public Info(String json) {
+            super(json);
+        }
+
+        /**
+         * Constructs an Info object using an already parsed JSON object.
+         * @param  jsonObject the parsed JSON object.
+         */
+        Info(JsonObject jsonObject) {
+            super(jsonObject);
+        }
+
+        /**
+         * Gets the user belonging to the group.
+         *
+         * <p>Note: Only id, name and login fields are available on the initial load.</p>
```

Comment:
Good thinking documenting this. Can we just rephrase this comment (as well as the others) to say something like:

"Note: the BoxUser.Info returned by this method will only have the ID, name, and login fields populated."